src/QueryPipeline/RemoteQueryExecutor.cpp

```diff
@@ -591,8 +591,8 @@ void RemoteQueryExecutor::finish()
     /// Send the request to abort the execution of the request, if not already sent.
     tryCancel("Cancelling query because enough data has been read");
 
-    /// If connections weren't created yet or query wasn't sent, nothing to do.
-    if (!connections || !sent_query)
+    /// If connections weren't created yet, query wasn't sent or was already finished, nothing to do.
+    if (!connections || !sent_query || finished)
```

Comment:
Query could be finished in tryCancel when we resume fiber to wait for current pending packet